### PR TITLE
Send document buffer when semantically highlighting old document versions

### PR DESCRIPTION
**Pull request by JoeRobich**
Enables Semantic Highlighting by default.

Before:
image

After:
image

### CODE REVIEW

**Comment by JoeRobich** [comment]
@nohwnd Please take a look

File: src/features/semanticTokensProvider.ts [hunk 1]
```diff
@@ -187,7 +187,11 @@ export default class SemanticTokensProvider extends AbstractProvider implements
             return null;
         }
 
-        let req = createRequest<protocol.V2.SemanticHighlightRequest>(document, new vscode.Position(0, 0));
+        const currentDocument = vscode.workspace.textDocuments.find(d => d.fileName === document.fileName);
+        const isCurrentVersion = currentDocument?.version === document.version;
+
+        // We need to include the document contents in our request when we are highlighting a version of the document other than the current version, such as in the Diff view.
+        let req = createRequest<protocol.V2.SemanticHighlightRequest>(document, new vscode.Position(0, 0), /* includeBuffer: */ !isCurrentVersion);
```
Comment (333fred) [inline src/features/semanticTokensProvider.ts:194]: How does this work? AFAICT, the service itself actually doesn't look at the text in the buffer: https://github.com/OmniSharp/omnisharp-roslyn/blob/ff21ac475c/src/OmniSharp.Roslyn.CSharp/Services/SemanticHighlight/SemanticHighlightService.cs#L27

**Comment by IrishBruse** [comment]
+1 this diffs became ugly recently

File: src/features/semanticTokensProvider.ts (hunk 1 above)
Comment (JoeRobich) [reply src/features/semanticTokensProvider.ts]: @333fred Updated to use new omnisharp request property instead of setting the buffer.

**Pull request merged**